qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -308,17 +309,79 @@ def handler(x):
                            " in template %d" %
                            (key, self._id, self._md_template.id))
 
-    def __setitem__(self, key, value):
-        r"""Sets the metadata value for the category `key`
+    def __setitem__(self, column, value):
+        r"""Sets the metadata value for the category `column`
 
         Parameters
         ----------
-        key : str
-            The metadata category
-        value : obj
-            The new value for the category
+        column : str
+            The column to update
+        value : str
+            The value to set. This is expected to be a str on the assumption
+            that psycopg2 will cast as necessary when updating.
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the column does not exist in the table
         """
-        raise QiitaDBNotImplementedError()
+        conn_handler = SQLConnectionHandler()
+
+        # try dynamic tables
+        sql = """SELECT EXISTS (
+                    SELECT column_name
+                    FROM information_schema.columns
+                    WHERE table_name=%s
+                        AND table_schema='qiita'
+                        AND column_name=%s)"""
+        exists_dynamic = conn_handler.execute_fetchone(
+            sql, (self._dynamic_table, column))[0]
+        # try required_sample_info
+        sql = """SELECT EXISTS (
+                    SELECT column_name
+                    FROM information_schema.columns
+                    WHERE table_name=%s
+                        AND table_schema='qiita'
+                        AND column_name=%s)"""
+        exists_required = conn_handler.execute_fetchone(
+            sql, (self._table, column))[0]
+
+        if exists_dynamic:
+            # catching error so we can check if the error is due to different
+            # column type or something else
+            try:
+                sql = """UPDATE qiita.{0}
+                         SET {1}=%s
+                         WHERE sample_id=%s""".format(self._dynamic_table,
+                                                      column)
+                conn_handler.execute(sql, (value, self._id))
+            except Exception as e:
+                column_type = conn_handler.execute_fetchone(
+                    """SELECT data_type
+                       FROM information_schema.columns
+                       WHERE column_name=%s AND table_schema='qiita'
+                    """, (column,))[0]
+                value_type = type(value).__name__
+
+                if column_type != value_type:
```

Comment:
This seems to be comparing SQL types (column_type) and python type (value_type). Is that kosher?